### PR TITLE
Fixes installation on PHP 8.1, Fixes CI errors, implements Vocalink 7.40 spec

### PR DESCRIPTION
**Pull request by JeremySkinner**
This PR resolves several issues:

- Implements the 7.40 spec
- Ensures the module can be installed on PHP 8.1 by adding missing php versions to composer.json
- Updates deprecated github actions
- Resolves issues with incompatible versions of composer being installed depending on php version (php 5.x requires composer version < 2.3) 
- Adds allowed plugins section to composer.json to allow installation with composer 2.x
- Migrates phpunit config to modern schema

Also includes the changes from #122 (implements Vocalink 7.30)

### CODE REVIEW

**Comment by JeremySkinner** [comment]
Updated to support the v7.40 spec

**Review by Olympic1 — CHANGES_REQUESTED** [review]

File: tests/Spec/DefaultSpecFactoryTest.php [hunk 1]
```diff
@@ -112,7 +112,10 @@ public function dataCreateAtDate()
             ['VocaLinkV640', new \DateTime('2020-10-12')],
             ['VocaLinkV640', new \DateTime('2021-10-04')],
             ['VocaLinkV680', new \DateTime('2021-10-05')],
-            ['VocaLinkV680', new \DateTime('2030-01-01')],
+            ['VocaLinkV680', new \DateTime('2022-07-24')],
+            ['VocaLinkV730', new \DateTime('2022-07-25')],
+            ['VocaLinkV740', new \DateTime('2023-06-05')],
```
[the remark this answers is not in the document]
Comment (JeremySkinner) [reply tests/Spec/DefaultSpecFactoryTest.php:118]: Added - thanks!

File: composer.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
         "issues": "https://github.com/cs278/bank-modulus/issues"
     },
     "require": {
-        "php": "^5.6 || ^7.2 || ^8.0",
+        "php": "^5.6 || ^7.2 || ^8.0 || ^8.1 || ^8.2",
```
Comment (cs278) [inline composer.json:19]: This change isn't needed `^8.0` covers all PHP 8 versions.

File: phpunit.xml.dist [hunk 3]
```diff
@@ -1,30 +1,20 @@
...
-            <exclude>
-                <file>src/Spec/VocaLinkV380/Generator.php</file>
-            </exclude>
-        </whitelist>
-    </filter>
-    <php>
-        <includePath>.</includePath>
-        <ini name="error_reporting" value="-1" />
-    </php>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" backupGlobals="false" backupStaticAttributes="false" colors="true" convertErrorsToExceptions="true" convertNoticesToExceptions="true" convertWarningsToExceptions="true" processIsolation="false" stopOnFailure="false" bootstrap="vendor/autoload.php" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd">
```
Comment (cs278) [inline phpunit.xml.dist:2]: I'll do this update when dropping PHP 5 support.

**Comment by cs278** [comment]
I think I've pulled in everything else from this PR, thanks @JeremySkinner and @Olympic1

**Pull request closed without merging**